topiary/predictor.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) 2017. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function, division, absolute_import
+import logging
+
+from .filters import (
+    apply_epitope_filters,
+    apply_effect_expression_filters,
+    apply_variant_expression_filters,
+    filter_silent_and_noncoding_effects,
+)
+from .sequence_helpers import (
+    protein_subsequences_around_mutations,
+    check_padding_around_mutation
+)
+from .epitope_prediction import (
+    build_epitope_collection_from_binding_predictions,
+)
+
+class TopiaryPredictor(object):
+    def __init__(
+            self,
+            mhc_model,
+            padding_around_mutation=None,
+            ic50_cutoff=None,
+            percentile_cutoff=None,
+            min_gene_expression=0.0,
+            min_transcript_expression=0.0,
+            only_novel_epitopes=False,
+            wildtype_ligandome_dict=None,
+            raise_on_error=True):
+        """
+        Parameters
+        ----------
+        mhc_model : mhctools.BasePredictor
+            Any instance of a peptide-MHC binding affinity predictor
+
+        padding_around_mutation : int
+            How many residues surrounding a mutation to consider including in a
+            candidate epitope. Default is the minimum size necessary for epitope
```

Comment:
where is this default set? In the constructor you just keep the passed-in value

Reply:
Clarified the default by moving `check_padding_around_mutation` call to `__init__`.